src/transformers/modeling_bart.py

```diff
@@ -801,7 +800,6 @@ def __init__(self, config: BartConfig):
         self.output_attentions = config.output_attentions
         self.output_hidden_states = config.output_hidden_states
 
-        padding_idx, vocab_size = config.pad_token_id, config.vocab_size
```

Comment:
Here it seems especially hard to remove the models dependency on the pad_token_id. @LysandreJik @sshleifer @julien-c

Reply:
yes we made a decision to avoid forcing the user to make a somewhat complicated mask for the decoder before calling forward, and instead make it for the user inside the model code. Making the mask requires knowing pad_token_id. Generate (which is on a branch) will additionally require knowledge of eos and bos.